### DISCUSSION TITLE
Hvordan spør vi brukere om dato?

### DISCUSSION THREAD

**Discussion by Febakke** [digdir/designsystemet#1681] [Patterns for public services]
How do we ask user for specific dates or a time period.

### Relevant
- When to use a date-picker

## Research
- https://github.com/digdir/innsiktsbibliotek/issues/22
- https://github.com/alphagov/govuk-design-system-backlog/issues/42
- [What makes an accessible date picker?](https://www.youtube.com/watch?v=D2Gy2WN4Iys) (Conference: Inclusive design)

**Comment by solhell** [comment]
Noen refleksjoner:
- Datepicker er bra, blant annet for å synliggjøre ukedager der det er viktig, men man bør ikke låse brukeren til å måtte benytte denne, slik Norwegian gjør. Datepicker er bra for mange, men for andre er det raskere å skrive inn datoen manuelt.
- Datepicker bør ha raske navigasjonsmuligheter mellom uker, måneder og år, f.eks. ved å bruke piltaster opp/ned for uker, start/slutt av linje for måneder og side opp/ned for år, i tillegg til knapper i grensesnittet for det samme.

**Comment by Camulos** [comment]
I think we should look at this pattern as several different things:

1. Input fields for dates, how should they work: 3 inputs vs 1 input with formatting? Should these inputs be able to interpret the users own formatting for dates? Should the interpreted date be written in a way that cannot be misunderstood (write the name of month vs only number).
2. Date-picker can be useful when you are selecting a date within the same month or within a short date-frame. What are the guidelines for these and what are the expected functionality for the technical component?
3. Data-range-picker can be useful when you are selecting a date range within 2-3 months. These can be confusing and should probably have some guidelines for offering predefined "often used" ranges.

In all there is different guidelines for the three and variants or different components needed.
